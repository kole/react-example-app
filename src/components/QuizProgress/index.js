import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const QuizProgressContainer = styled.div`
    font-size: 1.3em;
    font-weight: bold;
    padding: 2em 0;
`;

const QuizProgress = (props) => {
    const { currentQuestion, totalQuestions } = props;

    return (
        <QuizProgressContainer>{currentQuestion} of {totalQuestions}</QuizProgressContainer>
    );
}

QuizProgress.propTypes = {
    currentQuestion: PropTypes.any.isRequired,
    totalQuestions: PropTypes.any.isRequired
}

export default QuizProgress;
