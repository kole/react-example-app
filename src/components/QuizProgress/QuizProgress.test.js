/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import QuizProgress from './index.js';

describe('QuizProgress', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<QuizProgress currentQuestion={3} totalQuestions={10} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
