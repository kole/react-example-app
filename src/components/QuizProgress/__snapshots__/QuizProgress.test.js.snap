// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizProgress Snapshot 1`] = `
<div
  className="sc-bdVaJa gdYACt"
>
  3
   of 
  10
</div>
`;
